test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -282,7 +283,21 @@ protected <A extends Aggregator> A createAggregator(AggregatorFactories.Builder
      * Deprecated - this will be made private in a future update
      */
     @Deprecated
-    protected AggregationContext createAggregationContext(IndexSearcher indexSearcher, Query query, MappedFieldType... fieldTypes)
+    protected AggregationContext createAggregationContext(IndexReader indexReader, Query query, MappedFieldType... fieldTypes)
+        throws IOException {
+        return createAggregationContext(
+            indexReader,
+            createIndexSettings(),
+            query,
+            new NoneCircuitBreakerService(),
+            AggregationBuilder.DEFAULT_PREALLOCATION * 5, // We don't know how many bytes to preallocate so we grab a hand full
+            DEFAULT_MAX_BUCKETS,
+            false,
+            fieldTypes
+        );
+    }
+
+    private AggregationContext createAggregationContext(IndexSearcher indexSearcher, Query query, MappedFieldType... fieldTypes)
```

Comment:
Should this be marked as deprecated too?

Reply:
` * Deprecated - this will be made private in a future update`

I don't think so, this method will stay as we still need to construct those objects in order to run aggregations. It will be internal so we control the life cycle of those objects.